datafusion/expr/src/utils.rs

```diff
@@ -730,13 +730,10 @@ pub fn from_plan(
     inputs: &[LogicalPlan],
 ) -> Result<LogicalPlan> {
     match plan {
-        LogicalPlan::Projection(Projection { schema, .. }) => {
-            Ok(LogicalPlan::Projection(Projection::try_new_with_schema(
-                expr.to_vec(),
-                Arc::new(inputs[0].clone()),
-                schema.clone(),
-            )?))
-        }
```

Comment:
Here exist a bug, we shouldn't use original `schema`, because it can be changed.
Original code hidden some BUG.